docs/topics/testing/tools.txt

```diff
@@ -1362,6 +1364,22 @@ your test suite.
         with self.assertRaisesMessage(ValueError, 'invalid literal for int()'):
             int('a')
 
+.. method:: SimpleTestCase.assertWarnsMessage(expected_warning, expected_message, callable, *args, **kwargs)
+            SimpleTestCase.assertWarnsMessage(expected_warning, expected_message)
+
+    Asserts that execution of ``callable`` triggers ``expected_warning`` and
```

Comment:
Rather than repeating two paragraphs, I think this could simply say, "Analogous to SimpleTestCase.assertRaisesMessage but for assertWarnsRegex instead of assertRaisesRegex."

Also, add a `.. versionadded:: 2.1` annotation.